crypto/evp/p_lib.c

```diff
@@ -661,8 +661,14 @@ int EVP_PKEY_set_type_str(EVP_PKEY *pkey, const char *str, int len)
     return pkey_set_type(pkey, NULL, EVP_PKEY_NONE, str, len, NULL);
 }
 
+#ifndef OPENSSL_NO_DEPRECATED_3_0
 int EVP_PKEY_set_alias_type(EVP_PKEY *pkey, int type)
 {
+    if (!evp_pkey_is_legacy(pkey)) {
```

Comment:
Isn't this likely to  break existing applications that call this function?

Reply:
Uhm, depends on the internal key. What would you have this do if it's called with a provider-native key?

Reply:
I think I caught your thinking, and added an extra check to see if the key type is what the user asks for, and return 1 in that case.